sdk/servicebus/Azure.Messaging.ServiceBus/samples/DeadLetterQueue/Program.cs

```diff
@@ -0,0 +1,301 @@
+            ServiceBusReceiver deadletterReceiver = _client.CreateReceiver(queueName,
+                new ServiceBusReceiverOptions {SubQueue = SubQueue.DeadLetter});
+            while (true)
+            {
+                // receive a message
+                ServiceBusReceivedMessage message =
+                    await deadletterReceiver.ReceiveMessageAsync(TimeSpan.FromSeconds(10));
+                if (message != null)
+                {
+                    // write out the message deadletter information
+                    Console.WriteLine("Deadletter message:");
+                    Console.WriteLine($"DeadLetterReason = {message.DeadLetterReason}");
+                    Console.WriteLine($"DeadLetterErrorDescription = {message.DeadLetterErrorDescription}");
+
+                    // complete and therefore remove the message from the DLQ
+                    await deadletterReceiver.CompleteMessageAsync(message);
+                }
+                else
+                {
+                    // DLQ was empty on last receive attempt
+                    break;
+                }
+            }
+        }
+
+        private static Task ReceiveMessages(string queueName, CancellationToken cancellationToken)
+        {
+            var doneReceiving = new TaskCompletionSource<bool>();
+            ServiceBusProcessor processor = _client.CreateProcessor(queueName);
+
+            // close the receiver and factory when the CancellationToken fires 
+            cancellationToken.Register(
+                async () =>
+                {
+                    await processor.CloseAsync();
+                    doneReceiving.SetResult(true);
+                });
+
+            processor.ProcessMessageAsync += async args =>
+            {
+                ServiceBusReceivedMessage message = args.Message;
+                // If the message holds JSON data and the label is set to "Scientist", 
+                // we accept the message and print it.
+                if (message.Subject != null &&
+                    message.ContentType != null &&
+                    message.Subject.Equals("Scientist", StringComparison.InvariantCultureIgnoreCase) &&
+                    message.ContentType.Equals("application/json", StringComparison.InvariantCultureIgnoreCase))
+                {
+                    var body = message.Body;
+
+                    // System.Text.JSON does not currently support deserializing to dynamic
+                    Dictionary<string, string> scientist = body.ToObjectFromJson<Dictionary<string, string>>();
+
+                    lock (Console.Out)
+                    {
+                        Console.ForegroundColor = ConsoleColor.Cyan;
+                        Console.WriteLine(
+                            "\t\t\t\tMessage received: \n\t\t\t\t\t\tMessageId = {0}, \n\t\t\t\t\t\tSequenceNumber = {1}, \n\t\t\t\t\t\tEnqueuedTime = {2}," +
```

Comment:
I wonder if this would wrap when viewed through GitHub's normal restricted width....

Reply:
It appears to, not sure I follow.

Reply:
I made it shorter